From 3e1be6cb2889253791604388e3077720a84f79c3 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:42 -0400
Subject: [tools] perf: Fix pipe check regression in attr event callback

Message-id: <1410181752-25631-278-git-send-email-jolsa@redhat.com>
Patchwork-id: 91980
O-Subject: [PATCH RHEL7.1 BZ1134356 277/307] perf tools: Fix pipe check regression in attr event callback
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit a261e4a09a073451057e9dbe17783255ea94598d
Author: Jiri Olsa <jolsa@kernel.org>
Date: Thu Jun 5 18:51:44 2014 +0200

description
===========
The file factoring in builtin-inject.c object introduced regression
in attr event callback. The commit is:
  3406912 perf inject: Handle output file via perf_data_file object

Following hunk reversed the logic:
  -       if (!inject->pipe_output)
  +       if (&inject->output.is_pipe)

putting it back, following example now works:
  $ perf record -o - kill | perf inject -b | perf report -i -

Plus removing extra '&' (kudos to Arnaldo)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-inject.c b/tools/perf/builtin-inject.c
index 6a3af00..16c7c11 100644
--- a/tools/perf/builtin-inject.c
+++ b/tools/perf/builtin-inject.c
@@ -72,7 +72,7 @@ static int perf_event__repipe_attr(struct perf_tool *tool,
  if (ret)
   return ret;
 
- if (&inject->output.is_pipe)
+ if (!inject->output.is_pipe)
   return 0;
 
  return perf_event__repipe_synth(tool, event);
-- 
1.7.1